src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs

```diff
@@ -217,6 +217,8 @@ internal partial class RequestParameter
 
         public HttpParameterIn In => Protocol.Http is HttpParameter httpParameter ? httpParameter.In : HttpParameterIn.None;
         public bool IsFlattened => Flattened ?? false;
+
+        public bool IsApiVersion => Origin == "modelerfour:synthesized/api-version";
```

Comment:
maybe we could change it to use this `IsApiVersion` property as well in the code model converter: https://github.com/Azure/autorest.csharp/blob/a8860dd3653914481ae4f2831c3870672af1cebf/src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs#L134
so that we could only have one place with this magic string

Reply:
It's applied in the last commit :)